dockstore-webservice/src/main/java/io/dockstore/webservice/core/Checksum.java

```diff
@@ -55,4 +57,21 @@ public void setChecksum(String checksum) {
     public String toString() {
         return this.getType() + ":" + this.getChecksum();
     }
+
+    @Override
+    public boolean equals(final Object o) {
```

Comment:
Was a mssing equals  the reason for the odd behaviour?

Reply:
It was! 😭